src/bin/cargo/commands/check.rs

```diff
@@ -55,5 +57,15 @@ pub fn exec(config: &mut Config, args: &ArgMatches<'_>) -> CliResult {
     let compile_opts = args.compile_options(config, mode, Some(&ws), ProfileChecking::Unchecked)?;
 
     ops::compile(&ws, &compile_opts)?;
+
+    if features::nightly_features_allowed() {
```

Comment:
This will probably need an explicit opt-in.

Also, I'm a little concerned about the performance impact of adding this to the default set.  Have you checked to see how slow it is on medium-to-large libraries?

Is it possible that rustdoc could generate duplicate warnings from rustc?  I believe `missing_docs` would be duplicated, are there other warnings that could cause that?

Related to the two above issues, if this is to be included in the default set, then the logic would go in `filter_default_targets`.  Running the build logic twice would be a pretty big performance hit, and would cause cached warnings to be replayed twice, among other issues.  Unfortunately I think it could make the "duplicate errors" problem even more apparent, which makes me more concerned about having it in the default set.

Reply:
I didn't think about lint duplicates, but it's a good point. We can always allow the ones impacted to prevent this duplication. As for the compilation duplication, normally, rustdoc doesn't recompile anything. Or did I miss something?

Reply:
Personally I think this should be an opt-in with `cargo check --doc` or `cargo check --all-targets`.

> Is it possible that rustdoc could generate duplicate warnings from rustc? I believe missing_docs would be duplicated, are there other warnings that could cause that?

The only warnings rustdoc emits are here: https://github.com/rust-lang/rust/blob/a1a13b2bc4fa6370b9501135d97c5fe0bc401894/src/librustdoc/core.rs#L336-L350

I think of those `renamed_and_removed_lints` is the only one run in `rustc` other than `missing_docs`. I find the code around lints really hard to follow though, so I could be wrong. `-A missing_docs -A renamed_and_removed_lints` to rustdoc seems like a good idea.

> Have you checked to see how slow it is on medium-to-large libraries?

Slow. https://github.com/rust-lang/rust/issues/78761

Reply:
Since https://github.com/rust-lang/rust/pull/80527 I think cargo can solve the duplicate lint problem generally by passing `-A warnings -W rustdoc::all`.

Reply:
I guess that's subtly different because it also warns about lints that are allowed by default. So `-A missing_docs -A renamed_and_removed_lints -A unknown_lints` is probably better. https://github.com/rust-lang/rust/blob/8fd946c63a6c3aae9788bd459d278cb2efa77099/src/librustdoc/core.rs#L261-L267